### ISSUE TITLE
a11y: Fix keyboard navigation and focus display issues on the home screen

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12776]
a11y Issues: 62, 64, 65, 69, 60

Fix keyboard navigation and focus issues on the home screen:

**In the clusters table** - sorting icons do not support keyboard navigation or show focus - these should be fixed in the Sortable Table component.

![image](https://github.com/user-attachments/assets/ef0b0caa-3231-44d3-8b87-bd3563e5a530)

**Preferences link in the banner** - link does not support keyboard navigation or show focus

![image](https://github.com/user-attachments/assets/abf9c36e-e330-4ce9-b621-199b61030658)

**Banner Close Button** - close button does not support keyboard navigation or show focus

![image](https://github.com/user-attachments/assets/352cc601-4db9-4224-80ab-b33f86cb7cdb)

**Banner Image Close Button** - close button does not support keyboard navigation or show focus

![image](https://github.com/user-attachments/assets/24e6903a-08ac-4f69-83ea-8655ff0195b2)

** Issue 60 - Textual description missing for interactive image**

These two images are missing the `aria-label` attribute:

![image](https://github.com/user-attachments/assets/be65d4c7-3578-4aaa-8815-2c764615b312)
![image](https://github.com/user-attachments/assets/37d65b6f-fc3f-4e34-8c1b-5034bfe184d9)

**Comment by yonasberhe23** [comment]
Needs manual validation at this time as we dont have existing accessiblity tests. Follow-up ticket to create automated tests coverage https://github.com/rancher/qa-tasks/issues/1639

**Comment by aalves08** [comment]
@yonasberhe23 there's already an issue for creating automated tests: https://github.com/rancher/dashboard/issues/8986

where there's an associated PR with a WIP inclusion of cypress-axe: https://github.com/rancher/dashboard/pull/12900

We wanted to knock out the priority 0 issues before spending time on tests, since I imagine that's going to take a while to configure and run.

**Comment by yonasberhe23** [comment]
@aalves08 ah thanks for pointing that out. I'll close the ticket I created.

**Comment by IsaSih** [comment]
Navigation using tab and return work as expected in
Rancher | v2.11-1e40bc40c55a020fc67dc3c3dde47b41dc94caf1-head
Dashboard | master https://github.com/rancher/dashboard/commit/574248d1ce67bd5b8d3b6d95f716a979c22aaa9e
Rancher CLI | v2.11.0-rc.1
Helm | dev
Machine | v0.15.0-rancher125

However, the space bar does not perform click actions in elements inside the table. It works for elements outside tables.
Same as in https://github.com/rancher/dashboard/issues/12786, want to confirm @nwmac if this is working as expected? I reopened the issue because I believe that space should also perform the same actions as the return/enter keyboard shortcut.

**Comment by aalves08** [comment]
The decision made is that "links" that appear as buttons are to have the `space` key as trigger as simple links, anchors, non-button links should not have the `space` key as trigger, as per https://github.com/rancher/dashboard/pull/12959#discussion_r1901292826 + https://github.com/rancher/dashboard/pull/12959#discussion_r1901629859

**Comment by IsaSih** [comment]
I'm reopening this issue as I identified a regression from the version I tested last. 
The closing X icon of the image in the home screen is not being selected by tab navigation. It's selected by tab just if the user first clicks the image with the mouse. This is demonstrated in the screenshot:

https://github.com/user-attachments/assets/3204de6a-5d5c-4850-99e9-2d1c8e4e8c39
 
Tested in 
Rancher | v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
-- | --
Dashboard | master c987984
Rancher CLI | v2.11.0-rc.1

**Comment by aalves08** [comment]
> I'm reopening this issue as I identified a regression from the version I tested last. The closing X icon of the image in the home screen is not being selected by tab navigation. It's selected by tab just if the user first clicks the image with the mouse. This is demonstrated in the screenshot:
> 
>  Screen.Recording.2025-01-28.at.16.28.34.mov 
> Tested in
> 
> Rancher	v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
> Dashboard	master c987984
> Rancher CLI	v2.11.0-rc.1

Hi @IsaSih 

I forgot about this. Thanks for reporting it. 

I've opened https://github.com/rancher/dashboard/pull/13228 to make the close button always visible (it was only visible on mouse hover). To note that the home page banner is closeability is tied to a user preference as well, meaning that if that preference is set to `false` a user won't see the X to close it. I think it comes `true` (closable) as default. FYI @rak-phillip

**Comment by aalves08** [comment]
Since the solution is more complex, according to https://github.com/rancher/dashboard/pull/13228#issuecomment-2626795125 , I've created https://github.com/rancher/dashboard/issues/13251 to track the reported situation from @IsaSih .

Can this issue be closed or is this a blocker @IsaSih @nwmac ? FYI @rak-phillip

**Comment by IsaSih** [comment]
Closing the issue as the bug was tracked and fixed in a separate issue
